core/tests/test_utils.py

```diff
@@ -622,6 +622,9 @@ def process_and_flush_pending_tasks(self, queue_name=None):
                 if task.url == '/_ah/queue/deferred':
                     from google.appengine.ext import deferred
                     deferred.run(task.payload)
+                elif feconf.EMAILS_TASK_PREFIX in task.url:
+                    # Flush task if task is of emails taskqueue.
```

Comment:
Hi Sean, good news is that finally I solved this issue and now do not require this hack I made but bad news is that it broke a few tests so it will need some time to fix those. And after doing the fix system is working(i did manual backend testing through logs and its working).

Thanks for the help.